### DISCUSSION TITLE
How can i modify the Endpoint under wireguard mode

### DISCUSSION THREAD

**Discussion by hadesnth** [mitmproxy/mitmproxy#5943] [Q&A]
When i run mitmdump on my linux server
`mitmdump --listen-port 30001 --set block_global=false --mode wireguard`
then i generate
```
[Interface]
PrivateKey = kL85djwP2******************
Address = 10.0.0.1/32
DNS = 10.0.0.53

[Peer]
PublicKey = qpoZtmB******************
AllowedIPs = 0.0.0.0/0
Endpoint = 172.24.22.***:30001
```
but 172.24.22.*** is the LAN ip of my server, not the public network ip

how can i run it with the public network ip

**Comment by mhils** [comment]
The only purpose of `Endpoint` is for your client to find mitmproxy. You should be able to just change it to your preferred value in your WireGuard client. :)

**Reply by hadesnth** [reply]
i modify the endpoint as my server's public network ip, but it seems not work

**Reply by hadesnth** [reply]
My requirement is to run the wireguard mode of mitmproxy on the remote server, so that I can freely open the wireguard service on the mobile phone, so as to realize convenient packet capture anytime and anywhere.

I did not use the wireguard mode in the past, and directly filled in the server ip and port in the wifi http proxy on the iphone. This is feasible, but it is troublesome to fill in the ip and port every time.

**Reply by FahDon** [reply]
> i modify the endpoint as my server's public network ip, but it seems not work

Ensure your server can be reachable via public IP on port `30001` using `UDP` protocol.

**Comment by tomlawesome** [comment]
> i modify the endpoint as my server's public network ip, but it seems not work

How did you do this? I need to do the same but cannot find any where to edit the config. 

wireguard.conf only includes server keys as a json store, and I see no options in config.yaml either. 

I need to set the [Peer] endpoint specifically to work with a dynamic address. The MITM proxy is running on a network attached to LTE with no static IPv4, and I would rather it connect to my established site to site WG setup, which uses a FQDN/dynamic IP.
